test/testpulltask.py

```diff
@@ -0,0 +1,167 @@
+        if not self.sink:
+            return True
+
+        data = self.sink.status.get("github", None)
+        if not data:
+            return True
+        expected = data["status"]["description"]
+        context = data["status"]["context"]
+        statuses = github.statuses(self.sink.status["revision"])
+        status = statuses.get(context, None)
+        current = status.get("description", None)
+        if current and current != expected:
+            self.sink.status.pop("github", None)
+            self.sink.status.pop("badge", None)
+            self.sink.status.pop("irc", None)
+            sys.stderr.write("Verify collision: {0}\n".format(current))
+            return False
+        return True
+
+    def rebase(self):
+        try:
+            sys.stderr.write("Rebasing onto origin/master ...\n")
+            subprocess.check_call([ "git", "fetch", "origin", "master" ])
+            if self.sink:
+                master = subprocess.check_output([ "git", "rev-parse", "origin/master" ]).strip()
+                self.sink.status["master"] = master
+            subprocess.check_call([ "git", "rebase", "origin/master" ])
+            return None
+        except:
+            subprocess.call([ "git", "rebase", "--abort" ])
+            traceback.print_exc()
+            return "Rebase failed"
+
+    def stop_publishing(self, ret):
+        sink = self.sink
+        def mark_failed():
+            if "github" in sink.status:
+                sink.status["github"]["status"]["state"] = "failure"
+            if 'badge' in sink.status:
+                sink.status['badge']['status'] = "failed"
+            if "irc" in sink.status: # Never send success messages to IRC
+                sink.status["irc"]["channel"] = "#cockpit"
+        def mark_passed():
+            if "github" in sink.status:
+                sink.status["github"]["status"]["state"] = "success"
+            if 'badge' in sink.status:
+                sink.status['badge']['status'] = "passed"
+        if isinstance(ret, basestring):
+            message = ret
+            mark_failed()
+        elif ret == 0:
+            message = "Tests passed"
+            mark_passed()
+        else:
+            message = "{0} tests failed".format(ret)
+            mark_failed()
+        sink.status["message"] = message
+        if "github" in sink.status:
+            sink.status["github"]["status"]["description"] = message
+        del sink.status["extras"]
+        sink.flush()
+
+    def run(self, opts, github):
+
+        if self.ref:
+            subprocess.check_call([ "git", "fetch", "origin", self.ref ])
+            subprocess.check_call([ "git", "checkout", self.revision ])
+
+        # Split a value like verify/fedora-23
+        (prefix, unused, value) = self.context.partition("/")
+
+        os.environ["TEST_NAME"] = self.name
+        os.environ["TEST_REVISION"] = self.revision
+
+        if prefix == 'image':
+            os.environ["TEST_OS"] = 'fedora-23'
```

Comment:
as a follow-up we might want to resort to our default here, not hardcode fedora-23